source/monitoring/metrics-alerts/minio-metrics-and-alerts.rst

```diff
@@ -0,0 +1,365 @@
+
+.. metric:: minio_heal_objects_error_total
+
+   Objects for which healing failed in current self healing run
+
+.. metric:: minio_heal_objects_heal_total
+
+   Objects healed in current self healing run
+
+.. metric:: minio_heal_objects_total
+
+   Objects scanned in current self healing run
+
+.. metric:: minio_heal_time_last_activity_nano_seconds
+
+   Time elapsed (in nano seconds) since last self healing activity. This is set to -1 until initial self heal
+
+.. metric:: minio_inter_node_traffic_received_bytes
+
+   Total number of bytes received from other peer nodes.
+
+.. metric:: minio_inter_node_traffic_sent_bytes
+
+   Total number of bytes sent to the other peer nodes.
+
+.. metric:: minio_node_disk_free_bytes
+
+   Total storage available on a disk.
+
+.. metric:: minio_node_disk_total_bytes
+
+   Total storage on a disk.
+
+.. metric:: minio_node_disk_used_bytes
+
+   Total storage used on a disk.
+
+.. metric:: minio_node_file_descriptor_limit_total
+
+   Limit on total number of open file descriptors for the MinIO Server process.
+
+.. metric:: minio_node_file_descriptor_open_total
+
+   Total number of open file descriptors by the MinIO Server process.
+
+.. metric:: minio_node_io_rchar_bytes
+
+   Total bytes read by the process from the underlying storage system including cache, /proc/[pid]/io rchar
+
+.. metric:: minio_node_io_read_bytes
+
+   Total bytes read by the process from the underlying storage system, /proc/[pid]/io read_bytes
+
+.. metric:: minio_node_io_wchar_bytes
+
+   Total bytes written by the process to the underlying storage system including page cache, /proc/[pid]/io wchar
+
+.. metric:: minio_node_io_write_bytes
+
+   Total bytes written by the process to the underlying storage system, /proc/[pid]/io write_bytes
+
+.. metric:: minio_node_process_starttime_seconds
+
+   Start time for MinIO process per node, time in seconds since Unix epoc.
+
+.. metric:: minio_node_process_uptime_seconds
+
+   Uptime for MinIO process per node in seconds.
+
+.. metric:: minio_node_syscall_read_total
+
+   Total read SysCalls to the kernel. /proc/[pid]/io syscr
+
+.. metric:: minio_node_syscall_write_total
+
+   Total write SysCalls to the kernel. /proc/[pid]/io syscw
+
+.. metric:: minio_s3_requests_error_total
+
+   Total number S3 requests with errors
+
+.. metric:: minio_s3_requests_inflight_total
+
+   Total number of S3 requests currently in flight
+
+.. metric:: minio_s3_requests_total
+
+   Total number S3 requests
+
+.. metric:: minio_s3_time_ttbf_seconds_distribution
+
+   Distribution of the time to first byte across API calls.
+
+.. metric:: minio_s3_traffic_received_bytes
+
+   Total number of s3 bytes received.
+
+.. metric:: minio_s3_traffic_sent_bytes
+
+   Total number of s3 bytes sent
+
+.. metric:: minio_software_commit_info
+
+   Git commit hash for the MinIO release.
+
+.. metric:: minio_software_version_info
+
+   MinIO Release tag for the server
+
+.. _minio-metrics-and-alerts-alerting:
+
+Alerts
```

Comment:
Future Project - dedicated alerting tutorial with examples of configuring alert rules + alert manager to push to something like a NodeJS webhook or something similar.